### PR TITLE
Add rand(numRows: Int, numCols: Int) functions

### PR DESCRIPTION
**Pull request by xubo245**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)
## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

add rand(numRows: Int, numCols: Int) functions to DenseMatrix object,like breeze.linalg.DenseMatrix.rand()

### CODE REVIEW

**Comment by xubo245** [comment]
we can use it to replacebreeze.linalg.DenseMatrix.rand(numRows: Int, numCols: Int)

File: mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala [hunk 1]
```diff
@@ -497,6 +497,20 @@ object DenseMatrix {
...
+    * 
+    * @param numRows number of rows of the matrix
+    * @param numCols number of columns of the matrix
+    * @return DenseMatrix` with size `numRows` x `numCols` and values in U(0, 1)
+    */
+  @Since("2.0.0")
+  def rand(numRows: Int, numCols: Int): DenseMatrix = {
+    require(numRows.toLong * numCols <= Int.MaxValue,
+      s"$numRows x $numCols dense matrix is too large to allocate")
+    new DenseMatrix(numRows, numCols, Array.fill(numRows * numCols)((new Random).nextDouble()))
```
Comment (srowen) [inline mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala]: This makes a new RNG for every element, which isn't great.

Comment (xubo245) [inline mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala]: Can fix RNG, This makes a new RNG for all element :
    val rng = new Random()
    new DenseMatrix(numRows, numCols, Array.fill(numRows * numCols)(rng.nextDouble()))

**Comment by HyukjinKwon** [comment]
Doesn't that need a JIRA, test and PR descriptio maybe?

**Comment by xubo245** [comment]
I add test : https://github.com/apache/spark/pull/14424

**Comment by HyukjinKwon** [comment]
I guess it might be ncer if the contributions you make follow https://cwiki.apache.org/confluence/display/SPARK/Contributing+to+Spark.just like other contributions.

**Comment by xubo245** [comment]
@HyukjinKwon Thank you. 
This is my first time to push request to spark, Sorrry, I will follow the https://cwiki.apache.org/confluence/display/SPARK/Contributing+to+Spark  later.

**Comment by srowen** [comment]
Please don't open other PRs for related changes. Usually, you also need a JIRA. Although I understand the use for these methods, is there any use for them in Spark? we probably wouldn't add these things unless Spark itself needed them a few places, or else it was clearly a common task. I am not clear these are.

**Comment by xubo245** [comment]
@srowen sorry, please close the issue. I will learning more  before next PR. The PR is only because breeze have the function. In spark ,there is no use for them.
Could you tell me some issue for starter? Please.

**Comment by srowen** [comment]
@xubo245 you need to close the issues, we can't do so directly.

**Comment by xubo245** [comment]
ok

**Pull request closed without merging**